sklearn/tests/test_isotonic.py

```diff
@@ -346,3 +346,35 @@ def test_isotonic_zero_weight_loop():
 
     # This will hang in failure case.
     regression.fit(x, y, sample_weight=w)
+
+
+def test_fast_predict():
+    # test that the faster prediction (https://github.com/scikit-learn/scikit-learn/pull/6206)
+    # change doesn't affect out-of-sample predictions.
+    rng = np.random.RandomState(123)
+    N = 10**3
+    # X values over the -10,10 range
```

Comment:
Please add a `sample_weight` with some zeros in it to check that the duplicate removal is not impacted by the use of sample weights:

```python
sample_weight = rng.rand(n_samples)
sample_weight[sample_weight < 0.1] = 0
```